keanu-python/keanu/base.py

```diff
@@ -1,12 +1,23 @@
 class JavaObjectWrapper:
     def __init__(self, ctor, *args):
-        self._val = ctor(*args)
+        self._ctor = ctor
         self._args = args
+        try:
+            self._val = ctor(*args)
+        except TypeError as e:
+            raise EnvironmentError(
+                "Cannot call %s - this may be because the keanu jar has not been loaded into py4j's JVM's classpath" 
+                % JavaObjectWrapper.print_signature(ctor, args)
+            ) from e        
         self._class = self.unwrap().getClass().getSimpleName()
 
     def __repr__(self):
-        args = [str(arg) for arg in self._args]
-        return "[{0} => {1}: ({2})]".format(self._class, type(self), ",".join(args))
+        return "[{0} => {1}: {2}]".format(self._class, type(self), JavaObjectWrapper.print_signature(self._ctor, self._args))
+
+    @staticmethod
+    def print_signature(_ctor, _args):
```

Comment:
Make this private?

Reply:
I think I'm gonna need to nuke this anyway when I merge your PR in